### PR TITLE
[WIP] Allow external scoping

### PR DESCRIPTION
**Pull request by jonathantneal**
## Changes

- Provides short path to fixing CSS parsing issues.
- Provides short path to supporting container queries.
- This allows external style scoping with an `externalScoping` flag.
- Passes `transformOptions` to `preprocessStyle`, which should allow PostCSS to apply the styling.

## Testing

help needed

## Docs

help needed

### CODE REVIEW

File: cmd/astro-wasm/astro-wasm.go [hunk 1]
```diff
@@ -129,7 +135,7 @@ func preprocessStyle(i int, style *astro.Node, transformOptions transform.Transf
 		return
 	}
 	attrs := wasm_utils.GetAttrs(style)
-	data, _ := wasm_utils.Await(transformOptions.PreprocessStyle.(js.Value).Invoke(style.FirstChild.Data, attrs))
+	data, _ := wasm_utils.Await(transformOptions.PreprocessStyle.(js.Value).Invoke(style.FirstChild.Data, attrs, transformOptions))
```
Comment (matthewp) [inline cmd/astro-wasm/astro-wasm.go:138]: is this needed? I think that ProcessStyle has access to all of these options on the JS side anyways. Is there something that it needs?

Comment (jonathantneal) [reply cmd/astro-wasm/astro-wasm.go:138]: I think so. This was to have the object containing the hash value (e.g. `astro-hash1d2c3b4a`).

Comment (matthewp) [reply cmd/astro-wasm/astro-wasm.go:138]: We talked about this, make sense.

**Comment by natemoo-re** [comment]
@jonathantneal I'm still tracking this, just FYI. I really want to unblock this and allow power users to bring their own scoping.

**Comment by natemoo-re** [comment]
I dug into this over the weekend! 

While this is definitely one possible route we could take to unblock modern CSS features, the best fix would be to use a better CSS parser that supports all of the features we need out of the box. This PR would mean users need to bring their own logic for scoping, which I can see being a pain. 

I just opened a PR to swap in the `esbuild` CSS parser https://github.com/withastro/compiler/pull/329 Going to close this in favor of that.

**Pull request closed without merging**